### PR TITLE
SSD Miasma

### PR DESCRIPTION
**Pull request by Ty-Omaha**
**What does this PR do:**
Makes it easier to tell if someone is SSD - cool sprites by @Ionward 

**Images of sprite/map changes (IF APPLICABLE):**

### Update 2:

Should have further testing

🆑  Ty-Omaha and Ionward
add: Carbon humans (people, not animals) who go SSD will now get ZZZ bubbles to indicate they are SSD
/ 🆑

### CODE REVIEW

**Comment by Shadeykins** [comment]
Not a fan of the glowy blue or the white outlining (I would do plain Z's without outlines), but a good idea in principle for sure.

**Comment by Necaladun** [comment]
Like the ZZZs, like the idea, not sure about the blue glow thing...could it be made less intrusive/common? Maybe once every 5-10 seconds?

**Comment by Ty-Omaha** [comment]
@Necaladun 
>Like the ZZZs, like the idea, not sure about the blue glow thing...could it be made less intrusive/common? Maybe once every 5-10 seconds?

done

**Comment by Fox-McCloud** [comment]
the blue outline is gross, and the z's would be better if they were smaller and in a cloud.

People have internalized speech bubbles as being IC, for the most part; better to build on that.

File: code/modules/mob/living/login.dm [hunk 1]
```diff
@@ -6,8 +6,9 @@
 	update_sight()
 
 	//If they're SSD, remove it so they can wake back up.
+	if(player_logged && istype(src, /mob/living/carbon/human))
```
Comment (Fox-McCloud) [inline code/modules/mob/living/login.dm]: This is disgusting and an *egregious* violation of OOP. Don't do this.

File: code/modules/mob/living/logout.dm [hunk 2]
```diff
@@ -10,3 +10,5 @@
 			Sleeping(2)
 			player_logged = 1
 			last_logout = world.time
+			if(istype(src, /mob/living/carbon/human)) //only human carbon SSD warnings
```
Comment (Fox-McCloud) [inline code/modules/mob/living/logout.dm]: And again.

File: code/modules/mob/living/login.dm (hunk 1 above)
Comment (Ty-Omaha) [reply code/modules/mob/living/login.dm]: so what should I do instead?

File: code/modules/mob/living/logout.dm (hunk 2 above)
Comment (Ty-Omaha) [reply code/modules/mob/living/logout.dm]: so what should I do instead?

File: code/modules/mob/living/login.dm (hunk 1 above)
Comment (Fox-McCloud) [reply code/modules/mob/living/login.dm]: put it under the object that you want to have the behavior.

**Comment by Citinited** [comment]
I had this idea a while ago but never got far into implementing it. This will be really useful 👍 

I agree with Fox's suggestion of making this be a speech bubble thing, as it's a bit large as you have it now IMO

**Comment by Breenland** [comment]
I also support the ZZZ's being in a speech bubble, and this idea as a whole. No more excuses, don't touch the sleeping people!

**Comment by Kyep** [comment]
The "ZZZ" is best as a speech bubble.

The glow should be removed.

SSD people should still sleep / lay down. Having them display "ZZZ" while they're standing up looks odd.

**Comment by Ty-Omaha** [comment]
@Kyep the only reason he's standing up is because its in aghost

aghosting doesn't do this to you, this gif was taken before the ghost checks

**Comment by Kyep** [comment]
Maybe we want aghosting to do this to you, so people don't mess with our characters while they're aghosted.
If they put us in cryo.... and we stay there long enough to get despawned.... we can always respawn ourselves anyway.

File: code/modules/mob/living/carbon/human/login.dm [hunk 3]
```diff
@@ -1,4 +1,6 @@
 /mob/living/carbon/human/Login()
+	if(player_logged)
+		overlays -= image('icons/effects/effects.dmi', icon_state = "zzz_glow")
 	..()
```
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/human/login.dm]: I'd call `..()` first.

**Comment by Fox-McCloud** [comment]
Make sure the overlay stays attached to the player no matter what you do with them.

Dragging them around, shoving them into a locker, a sleeper, what have you.

Also have someone go SSD when inside something and have them pop out of that thing, too.

**Comment by Ty-Omaha** [comment]
@Fox-McCloud interesting, I took the ghosting logic from the old proc (which I tested) and put it into the new proc

File: code/modules/mob/living/carbon/human/login.dm (hunk 3 above)
Comment (Ty-Omaha) [reply code/modules/mob/living/carbon/human/login.dm]: Doing that would reset player_logged

**Comment by Ty-Omaha** [comment]
@Fox-McCloud  Did you test it? I just did and it's not doing either of those things



https://i.imgur.com/XaqplJ8.gifv

**Comment by Fox-McCloud** [comment]
Yeah, I did.

**Comment by Ty-Omaha** [comment]
Weird.

**Comment by MrMagolor** [comment]
Really, why wouldn't people that are normally unconscious snore too?

**Comment by Citinited** [comment]
Because the point is to tell at a glance if someone is actually SSD

**Comment by Ty-Omaha** [comment]
@variableundefined What update is needed? I cannot reproduce anything @Fox-McCloud says happens. Even tested it deadminned:

**Comment by farie82** [comment]
Found an issue:
**Problem Description**:
If a ssd person dies and becomes a blob zombie he keeps the miasma

**What did you expect to happen**: 
He losing the miasma once zombified

**What happened instead**:
He kept the miasma

**Why is this bad/What are the consequences:**
People will hesitate to kill these zombies due to the ssd signs. BWOINK

**Steps to reproduce the problem**:
1. Go ssd
2. Die
3. Become a blob zombie

**Comment by TDSSS** [comment]
I think it might just be all dead people
I've seen a dead chaplain get found in maint once and he still had an SSD miasma.

**Comment by Ty-Omaha** [comment]
Okay so I have narrowed it down.

If the person just logs out while dead instead of ghosting they will have the overlay.

I will experiment with a stat check and using the "this person is SSD!" flavor text code check instead later today and update the sprite

**Comment by Ty-Omaha** [comment]
Also to test later: does a ghost disconnecting cause the body to have the overlay

**Comment by Ty-Omaha** [comment]
Ghosts disconnecting does not cause the overlay to show

**Comment by Ty-Omaha** [comment]
added dead check and removed glow

**Comment by variableundefined** [comment]
I'll try to review this as soon as possible.

**Pull request merged**